lib/evmone/instructions_calls.cpp

```diff
@@ -180,6 +180,84 @@ evmc_status_code create_impl(StackTop stack, ExecutionState& state) noexcept
     return EVMC_SUCCESS;
 }
 
+evmc_status_code create3(StackTop stack, ExecutionState& state, code_iterator& pos) noexcept
+{
+    if (state.in_static_mode())
+        return EVMC_STATIC_MODE_VIOLATION;
+
+    const auto initcontainer_index = uint8_t{pos[1]};
+
+    const auto& container = state.original_code;
+    const auto eof_header = read_valid_eof1_header(state.original_code);
+    const auto initcontainer = eof_header.get_container(container, initcontainer_index);
+
+    const auto endowment = stack.pop();
+    const auto input_offset_u256 = stack.pop();
+    const auto input_size_u256 = stack.pop();
+    const auto salt = stack.pop();
+
+    stack.push(0);  // Assume failure.
+    state.return_data.clear();
+    state.deploy_container.reset();
+
+    // Charge for initcode validation and hashing.
+    const auto initcode_word_cost = 8;
+    const auto initcode_cost = num_words(initcontainer.size()) * initcode_word_cost;
+    if ((state.gas_left -= initcode_cost) < 0)
+        return EVMC_OUT_OF_GAS;
+
+    if (!check_memory(state, input_offset_u256, input_size_u256))
+        return EVMC_OUT_OF_GAS;
+
+    const auto input_offset = static_cast<size_t>(input_offset_u256);
+    const auto input_size = static_cast<size_t>(input_size_u256);
+
+    // Charge for input data hashing.
```

Comment:
We should not hash input so this is not needed.

Reply:
Why not? I think it's useful (calling constuctor with different arguments normally means someone wants to deploy another instance at diferrent address) and mimics the way CREATE2 works.

Reply:
Found the discussion where some people are in favor of not hashing arguments https://github.com/ethereum/eips/pull/3171